freqtrade/freqai/freqai_interface.py

```diff
@@ -467,7 +467,7 @@ def data_cleaning_predict(self, dk: FreqaiDataKitchen, dataframe: DataFrame) ->
         if ft_params.get(
             "principal_component_analysis", False
         ):
-            dk.pca_transform(dataframe)
+            dk.pca_transform(self.dk.data_dictionary['prediction_features'])
```

Comment:
Doesn't seem like the dataframe argument is used in this function ...

as it's used in base models - it might be people pass in something and expect that to be used (instead of prediciton_features) (basically a dataframe that's not equal to the one in data_dictionary) - and then get surprising results.
maybe we should remove that argument - if we assume "dataframe magic" to happen in the background... ?